pkg/common/configs/configvalidator.go

```diff
@@ -213,9 +213,12 @@ func checkLimitResource(cur QueueConfig, users map[string]map[string]*resources.
 func checkQueueMaxApplications(cur QueueConfig) error {
 	var err error
 	for _, child := range cur.Queues {
-		if cur.MaxApplications != 0 && (cur.MaxApplications < child.MaxApplications || child.MaxApplications == 0) {
+		if cur.MaxApplications != 0 && cur.MaxApplications < child.MaxApplications {
 			return fmt.Errorf("parent maxApplications must be larger than child maxApplications")
 		}
+		if cur.MaxApplications != 0 && child.MaxApplications == 0 {
+			return fmt.Errorf("maxApplications is either undefined or zero, which is not allowed when parent queue's maxApplications is defined")
+		}
```

Comment:
Does this mean that whenever a parent has maxApps defined, all the child queues are required to have this setting too?

If that's the case, I'm -1 on this and 100% against this change. Imagine "root.dev" having maxapps with a bunch of parent queues for teams and leaf queues for two dozen developers. Once you set it, you have to define for the entire hierarchy.

Reply:
> Does this mean that whenever a parent has maxApps defined, all the child queues are required to have this setting too?

This PR aims to add more meaningful message to inspire users how to fix configs. The behavior you described is already there, and I had filed a ticket to discuss such weird behavior. https://issues.apache.org/jira/browse/YUNIKORN-2258

Reply:
I see. We must get rid of this then. Don't make this change to the config validator and remove the existing logic which performs the runtime checks.

Reply:
Added my thoughts to YUNIKORN-2258.

Reply:
@pbacsko thanks for feedback. Do you approve of this PR? I'd like to merge it before YUNIKORN-2258 :)

Reply:
@chia7712 my stance is that we don't need this check. If we submit it, then it has to be reverted in YUNIKORN-2258. So I don't see the point of adding it.

Reply:
> my stance is that we don't need this check. 

Sorry that I neglected your comment before.

> If we submit it, then it has to be reverted in YUNIKORN-2258. So I don't see the point of adding it.

The reasons I want to merge it are shown below.

1. YUNIKORN-2258 hasn't yet reached consensus, so it may still open in next release. Hence, the error message added by this PR can give more readable output when users encounter such issue.

2. The changes introduced by this PR is small, so it wouldn't burn out our servers 😀